scripts/dataflow_write_file_info.py

```diff
@@ -10,6 +10,32 @@
 
 from config import ssh_args, db_args
 
+
+# get deduplicated log_sensor_file table from database
+def _dedup_log_sensor_file(sensor_file_df):
+    '''returns the deduplicated log_sensor_file table
+    
+        context:
+        Sensors such as intel have duplicate sensor_file_paths but unique
+        sensor_file_ids for depth vs rgb - same holds true for mbients that
+        have one data file but two sensor_file_ids for accelerometer and
+        gyroscope. Thus we can have duplicate sensor files over two rows.
+        We want to write one line in the log_file table per device hence
+        this function.
+    '''
+    def __concat_list_elements(x):
+        c=''
+        for i in x:
+            c = c+str(i)
+        return c
+    # Convert an array of filenames into a single concatenated string. If order of filename array
```

Comment:
Could this be addressed by sorting here?  Assuming the file name list isn't too long it might not be expensive.

Reply:
The table structure is such that the file names are a list in a single cell - i.e. postgresql datatype of text[]. The list itself contains very few file names, often just one. For any sort operation in the filenames, they will still need to be unpacked. I chose this route since the apply() and drop_duplicates() methods are efficiently implemented in pandas. I think sorting on another column and then selecting unique lists of file names would also work.